### PR TITLE
Clean up TODOs and FIXMEs

### LINKED ISSUE #1365: Go through TODOs

### PR DESCRIPTION
**Pull request by f0uriest**
Renames all FIXMEs to TODOs, and removes some old ones that are either already done or are not planned.

At the next dev/triage meeting we should go through the remaining ones and either make actual issues for them or remove them.

Resolves #1365

### CODE REVIEW

File: desc/optimize/optimizer.py [hunk 1]
```diff
@@ -198,7 +198,6 @@ def optimize(  # noqa: C901 - FIXME: simplify this
             eq_params_init = eq.params_dict.copy()
 
         options = {} if options is None else options
-        # TODO: document options
```
Comment (YigitElma) [inline desc/optimize/optimizer.py:201]: this could have been nice

File: tests/test_bootstrap.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def test_trapped_fraction_analytic(self):
             modB = np.where(
                 mask, 9.0 + 3.7 * np.sin(theta - NFP * zeta), 13.0 + 2.6 * np.cos(theta)
             )
-            # todo: find value for sqrt_g_r value to test axis limit
+            # TODO: find value for sqrt_g_r value to test axis limit
```
Comment (YigitElma) [inline tests/test_bootstrap.py:89]:
```suggestion
            # TODO: find value for sqrt_g_r to test axis limit
```

**Review by YigitElma — DISMISSED** [review]
Good for now, we can clean-up the rest later

**Review by unalmis — APPROVED** [review]

**Review by YigitElma — APPROVED** [review]

File: desc/optimize/optimizer.py (hunk 1 above)
Comment (dpanici) [reply desc/optimize/optimizer.py:201]: I hope what I did works to add a link to the right page in the docs

**Review by dpanici — DISMISSED** [review]

File: desc/optimize/optimizer.py [hunk 3]
```diff
@@ -137,7 +137,9 @@ def optimize(  # noqa: C901
             Maximum number of iterations. Defaults to 100.
         options : dict, optional
             Dictionary of optional keyword arguments to override default solver
-            settings. See the code for more details.
+            settings. See the documentation page ``Optimizers Supported`` for
+            more details (https://desc-docs.readthedocs.io/en/stable/optimizers.html)
+            to check the options for each specific available optimizer.
```
Comment (YigitElma) [inline desc/optimize/optimizer.py:142]: Thanks!

**Review by YigitElma — APPROVED** [review]

File: desc/optimize/optimizer.py (hunk 1 above)
Comment (YigitElma) [reply desc/optimize/optimizer.py:201]: Thanks!

**Pull request merged**